nsv13/code/modules/munitions/ship_weapons/energy_weapons/phaser.dm

```diff
@@ -107,6 +169,20 @@
 		return FALSE
 	if(charge < charge_per_shot*shots) //Do we have enough ammo?
 		return FALSE
+	if(weapon_state == STATE_OVERLOAD) //have we overheated?
+		return FALSE
+	if(weapon_state == STATE_VENTING) //are we venting heat?
+		return FALSE
+	if(freq <=10) //is the frequincy of the weapon high enough to fire?
+		overload()
+		return FALSE
+	if(alignment == 0)
+		for(var/mob/living/M in get_hearers_in_view(7, src)) //burn out eyes in view
+			if(M.stat != DEAD && M.get_eye_protection() < 2) //checks for eye protec
+				M.flash_act(10)
+				to_chat(M, "<span class='warning'>You have a second to watch the casing of the gun glow a dull red before it erupts in a blinding flash as it self-destructs</span>")   // stealing this from the plasmagun as well
+		explosion(get_turf(src), 0, 1, 3, 5, flame_range = 4)
+		overload()
 	else
 		return TRUE
```

Comment:
What actually does this do?

Reply:
base return value is null, which, while similar to FALSE, is not *exactly* FALSE. If a proc always returns TRUE / FALSE, it should generally do that for all cases for consistency's sake.
Consequently, with that present, since any `if` that is fulfilled returns without reaching the end, the else is not neccessary.